### PR TITLE
11348 refactor mpi find profile

### PR DESCRIPTION
**Pull request by bosawt**
## Summary

- This PR refactors the main MPI service `find_profile` function so that it takes parameters as discrete attributes, rather than a full `UserIdentity` object. This should make it much simpler to invoke and use the MPI querying functionality. The functionality itself isn't a huge change, but since it touches so much of the codebase, the breadth of the change is absolutely massive

## Related issue(s)
- department-of-veterans-affairs/vets-api#11348

## Testing done

- Authenticated, confirmed MPI was successfully queried and results returned
- Tested with mocked errors and confirmed that errors were propagated as expected
- Tested various functionality in the site that requires MPI querying functionality

## What areas of the site does it impact?
- Basically everything

## Acceptance criteria

- [ ]  Will need multiple teams to inspect MPI functionality based on their own interactions and confirm they continue to work as expected with the changes from this PR

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb [hunk 1]
```diff
@@ -1072,18 +1072,8 @@ def obj.class
           }
         end
         let(:mvi_profile) { build(:mvi_profile) }
-        let(:mvi_profile_response) do
-          MPI::Responses::FindProfileResponse.new(
```
Comment (stiehlrod) [inline modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb:1076]: Is there a reason to use create, rather than build?

File: modules/claims_api/spec/requests/v1/power_of_attorney_request_spec.rb [hunk 2]
```diff
@@ -115,7 +115,7 @@
             end
 
             context 'when consumer is Veteran and missing EDIPI' do
-              it 'catches a raised 422' do
```
Comment (stiehlrod) [inline modules/claims_api/spec/requests/v1/power_of_attorney_request_spec.rb:118]: If we change the description, shouldn't we change the response status as well, or vice versa? but then this test will fail...

Comment (bosawt) [reply modules/claims_api/spec/requests/v1/power_of_attorney_request_spec.rb]: Oops, good catch. I originally had this raising a `500`, but decided instead to add the error capture of `MPI::Errors::ArgumentError` in the Application Controller, to try to preserve the functionality as closely as possible. So the spec should say `catches a raised 422`, I just changed the description back to that

File: modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb (hunk 1 above)
Comment (bosawt) [reply modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb:1076]: Since I was changing that other spec I decided just to change this to `build` for these. Doesn't appear to have an effect and it's prob better practice as a general rule of thumb to prefer `build` unless `create` is necessary

**Review by rjohnson2011 — APPROVED** [review]
Now that's what I call a PR. We'll coordinate at 3pm EST to bypass the danger limit check and get this in. Great work!
